### DISCUSSION TITLE
add user configurable decimal precision to thermostat card (and more-info)

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#18710] [Dashboard]
https://www.home-assistant.io/dashboards/thermostat/

please also add this for Tile card

precision currently is 0.1.
this is rather a nasty problem for thermostats using a precision of 0.5, like the Plugwise I am using.

there is a certain lag in communication  back and forth, so sliding the slider, causes jumping around of the numbers, probably caused by the fact the actual device doesnt use these 0.1 precision settings in the first place

several custom cards allow us to set a precision (step), and my request is to do so too in the core HA thermostat card, inherited by the new more-info popup too

![Nov-21-2023 09-18-34](https://github.com/home-assistant/frontend/assets/33354141/3a1c373c-4e34-4cd8-9121-1d11192aa330)

example custom card:
```
  - type: custom:numberbox-card
    entity: climate.voorkamer
    min: 5
    max: 21
    step: 0.5
```
or

```
  - type: custom:simple-thermostat
    entity: climate.voorkamer
    step_size: 0.5
```

thanks for considering

**Comment by karwosts** [comment]
I thought the step size comes from the climate attribute `target_temp_step`. Does your entity have this at 0.1? It's supposed to default to 0.5 for Celsius.

**Reply by Mariusthvdb** [reply]
not sure about that tbh, might need to chime in @bouwew for that, he is the expert on that integration
I guess this inserts that: https://community.home-assistant.io/t/plugwise-core-and-custom-component/236250/366?u=mariusthvdb
but, besides that, an optional step_size in the card would be very handy to overcome those integrations using 0.1 step.

**Reply by karwosts** [reply]
Other idea would be to `customize:` the target_temp_step if you want to override the step size. 

Yes a card could implement a custom step size property, but that would not affect the more-info (which doesn't have any mechanism to customize it), so the experience would be inconsistent.

**Reply by Mariusthvdb** [reply]
Never thought of that customize option, sounds interesting ! Let me try that , cool

**Comment by bouwew** [comment]
The step-size reported by various Plugwise devices is 0.1, after discussing with @frenck it was decided to keep this value.
Also because now the plus- and minus-buttons in the more-info card allow for an easier setting of the temperature setpoint.

**Reply by Mariusthvdb** [reply]
Exactly. And though it admittedly is not difficult, it is rather a nuisance to have to click 10 times for 1 degree up or down . Let alone for bigger manual changes .

Hence the request to allow user configurability in the card on step size

**Comment by Mariusthvdb** [comment]
I did

```
    climate.voorkamer:
      target_temp_step: 0.5
```

and look at that:

![Nov-23-2023 22-51-46](https://github.com/home-assistant/frontend/assets/33354141/0f2e2d3a-75fc-4b77-9e9e-68c5e6aafbdd)

which is wonderful.

however, the thermostat more-info is still unusable :

![Nov-23-2023 22-51-04](https://github.com/home-assistant/frontend/assets/33354141/9846cea2-d9b4-4d7a-89f7-1724f429953a)

**Reply by karwosts** [reply]
Only reason I can think it would jitter like that is if in response to the service call your entity is reporting a state change, and reporting its state as the old value. 

Typically I would expect:
- press the button
- set_temperature service is called
- some small delay
- entity updates state with new temperature.

**Reply by Mariusthvdb** [reply]
we did see this jitter behavior before on certain switches, where the toggle would go back and forth, making that interaction utterly unreliable, simply because the user would respond to that.

it would be better if the slider would remain at the spot the user releases it, wait for the device to respond and then settle.

in its current state, the new more-info cant be used as a serious interface, at least not for the various heaters I am using, Plugwise and Mill heaters integration in core HA
